File: commandPattern/invoker.cpp
#include <iostream>
#include "./invoker.h"
#include "./command.h"

void SubmitButton::SetCommand(Command *command)
{
	_command = command;
}

void SubmitButton::Submit()
{
	std::cout << "Ready to submit" << std::endl;
	if (_command != 0)
	{
		_command->Execute();
	}
}

File: commandPattern/csharp/Invoker.cs
class SubmitButton
{
	private Command? _command;

	public void SetCommand(Command command)
	{
		_command = command;
	}

	public void Submit()
	{
		System.Console.WriteLine("Ready to submit");
		if (_command is Command)
		{
			_command.Execute();
		}
	}
}